torch/distributed/checkpoint/state_dict.py

```diff
@@ -162,7 +165,8 @@ def _get_fqns(model: nn.Module, name: str, skip_ddp_prefix: bool = True) -> FQNS
                 flat_param = getattr(curr_obj, FLAT_PARAM)
                 if prefix:
                     prefix = f"{prefix}."
-                return {f"{prefix}{fqn}" for fqn in flat_param._fqns}
+                fqns = {f"{prefix}{fqn}" for fqn in flat_param._fqns}
+                return {fqn.replace(_CHECKPOINT_PREFIX, "") for fqn in fqns}
```

Comment:
We should do this for all parallelisms. Some people use activation checkpoint with DDP.